HACKING.md

```diff
@@ -108,8 +112,55 @@ otherwise an official image can be run with
 
     $ make run
 
+Log in to the FAF container
+
+    $ make sh
+
+Set up the database in the `[Storage]` section of `/etc/faf/faf.conf` as follows:
```

Comment:
Could this be part of `run_local` action?

Reply:
Not sure if I understand. Do you mean to automate the config file manipulation by adding stuff to the `run_local` target in`docker/Makefile`?  
  
The config file `/etc/faf/faf.conf` has to be edited like this no matter if you've run `make run` or `make run_local` so I thought it made sense to put this section after both of these options are mentioned, rather than bundling it with one of them.

Reply:
> Do you mean to automate the config file manipulation by adding stuff to the `run_local` target in`docker/Makefile`?

Yes.

> The config file `/etc/faf/faf.conf` has to be edited like this no matter if you've run `make run` or `make run_local`

Oh, then do it for both.

Reply:
See now. I added one Makefile target to use along with what was `make run_local`. With `make run`, it doesn't work in the current form as the container is run as non-root, so you still have to do it by hand.

Reply:
You don't have to edit the `/etc/faf/faf.conf` with the non-root container. The container is ran under user `faf` and there are `env` variables set to connect to the database (PGUSER, PGHOST, ...).

We could add this to Dockerfile_local and you wouldn't need to do the config changes.
```
echo 'Defaults    env_keep =  "PGHOST PGUSER PGPASSWORD PGPORT PGDATABASE"' >> /etc/sudoers.d/faf
```

Reply:
Pardon my being thick but where exactly in `Dockerfile_local` should I add this? I couldn't get it to work.  
  
Also, is there a similar way to set up the redis container in `/etc/faf/plugins/celery_tasks.conf`? If not, is there a way to get root in the non-root FAF container (run with `make run`, rather than `make run_local`) so I can edit the file by hand?

Reply:
> Pardon my being thick but where exactly in `Dockerfile_local` should I add this? I couldn't get it to work.

Below/Above the `sed` is probably the right place. (add `&& \` at the end)
https://github.com/abrt/faf/blob/92cfd721908756383965e429ef10676c2a02f350/docker/Dockerfile_local#L36
___
> Also, is there a similar way to set up the redis container in `/etc/faf/plugins/celery_tasks.conf`? If not, is there a way to get root in the non-root FAF container (run with `make run`, rather than `make run_local`) so I can edit the file by hand?

You will have to do the same trick as with the `postgres` and replace:
https://github.com/abrt/faf/blob/92cfd721908756383965e429ef10676c2a02f350/src/pyfaf/celery_tasks/__init__.py#L14
with:
https://github.com/abrt/faf/blob/92cfd721908756383965e429ef10676c2a02f350/src/pyfaf/common.py#L179
and then supply the env vars (`RDSBROKER`, `RDSBACKEND` or w/e) in:
https://github.com/abrt/faf/blob/92cfd721908756383965e429ef10676c2a02f350/docker/Makefile#L16
___
And maybe change the openshift template:
https://github.com/abrt/faf/blob/master/docker/openshift/faf-template.yml

Reply:
Reworked, HACKING.md rewritten accordingly.